### ISSUE TITLE
GPO or reg-settings with security groups issue

### ISSUE THREAD

**Issue by IwistIT** [pseymour/MakeMeAdmin#30]
I am trying to regulate the use of this nice application with the GPO's \ reg-keys you set up for this. If I enter a user account or SID for the Allowed Entities then it works fine.

However, when an AD group or AD group SID is entered here, the users are not authorized to use the application.
Can you help me with this?

Thanks in advance.

Willem

**Comment by pseymour** [comment]
Could you post a screenshot of your registry settings? If you don't want to post it publicly, send to my same user name at protonmail.com.

**Comment by Luckyson666** [comment]
Hi, I have the same experience. My registry settings on test workstation:

![image](https://user-images.githubusercontent.com/80003780/109864319-f8ef3e80-7c62-11eb-88b5-8f5a198260e4.png)

Could you help us with it ?

Thank you.

**Comment by IwistIT** [comment]
Hi Luckyson666,

I have this experience only on Azure AD joined devices, that the group does not work, but directly added users does. The group does work on domain-joined devices.

**Comment by Luckyson666** [comment]
Hi IwistIT, today I tried the same as yesterday, I created new on-premise security group and updated GPO with the new group name - and now it's working as expected, I don't understand the behaviour, because I did exactly the same as before, maybe some my fault "somewhere" :)

We are using hybrid AD join environment and I thought that I would just easily add SinclairMakeMeAdmin.admx to Configuration profiles to Intune, but I was not able to create a functional OMA-URI for this custom admx in Intune. Did you succeed with the deploy Make Me Admin settings via Intune ?

**Comment by IwistIT** [comment]
Yes, I did:

First you have to import the policy definition:
![image](https://user-images.githubusercontent.com/7647984/110100158-605fd800-7da2-11eb-9e0e-87974a0aadb2.png)

OMA-URL I used for import:
./Device/Vendor/MSFT/Policy/ConfigOperations/ADMXInstall/MakeMeAdmin/Policy/SinClairMakeMeAdminAdmx

[MakeMeAdminCSPDefinition.txt](https://github.com/pseymour/MakeMeAdmin/files/6089661/MakeMeAdminCSPDefinition.txt)

After import you can find them in the registry: 

![image](https://user-images.githubusercontent.com/7647984/110103240-fea16d00-7da5-11eb-8e76-ba1133598ae5.png)


And then use these definitions to to define the policy-setting:

![image](https://user-images.githubusercontent.com/7647984/110101638-1972e200-7da4-11eb-8e53-a1579d335506.png)

![image](https://user-images.githubusercontent.com/7647984/110103392-30b2cf00-7da6-11eb-80a9-06e3aa623dea.png)


![image](https://user-images.githubusercontent.com/7647984/110101927-72db1100-7da4-11eb-8745-28f1e8b2a93b.png)

![image](https://user-images.githubusercontent.com/7647984/110103716-87b8a400-7da6-11eb-8278-8a3489d94607.png)


And so on for the other settings

**Comment by Luckyson666** [comment]
Hi IwistIT, great, thank you, I'll try it and let you know here ;)

**Comment by Luckyson666** [comment]
Hi IwistIT, your configuration works. And I see now the problem I had in my OMA-URI configuration. I tried to use the original admx file with the value:

<parentCategory ref="makemeadmin:MakeMeAdmin" />

You changed it to:

<parentCategory ref="MakeMeAdmin" />

So thank you again !  ;)

**Comment by IwistIT** [comment]
Thanks for your feedback. It's great that the CSP is now working. You're welcome.

**Comment by CloudViking86** [comment]
I am really scratching my head here, trying to get the exact same thing to work using Intune against an AAD-joined device.
I've tried the example above and replacing occurrences of:
`<parentCategory ref="scc:SinclairRoot" /> `
and
`<parentCategory ref="makemeadmin:MakeMeAdmin" />`
with
`<parentCategory ref="MakeMeAdmin" />`

Now bear with me here, the first time I am working with ADMX-files and deploying them through Intune:
* There are two ADMX-files;
SinclairBase.admx;
https://github.com/pseymour/MakeMeAdmin/blob/master/Setup/GroupPolicy/SinclairBase.admx
SinclairMakeMeAdmin.admx;
https://github.com/pseymour/MakeMeAdmin/blob/master/Setup/GroupPolicy/SinclairMakeMeAdmin.admx

In Intune:

===

Name: "MakeMeAdmin ADMX"
which contains both below:

Name: SinclairBase
Description: <None>
OMA-URI:` ./Device/Vendor/MSFT/Policy/ConfigOperations/ADMXInstall/MakeMeAdmin/Policy/SinclairBase`
Data type: String
Value: <content of SinclairBase.admx>

Name: SinclairMakeMeAdmin
Description: <None>
OMA-URI:` ./Device/Vendor/MSFT/Policy/ConfigOperations/ADMXInstall/MakeMeAdmin/Policy/SinclairMakeMeAdmin`
Data type: String
Value: <content of SinclairMakeMeAdmin.admx>

As per the legend on how to construct an OMA-URI for an ADMX-file (for ingestion);
`./Vendor/MSFT/Policy/ConfigOperations/ADMXInstall/{AppName}/{SettingType}/{AdmxFileName}`
{AppName} = MakeMeAdmin
{SettingType} = Policy
{AdmxFileName} = SinclairBase OR SinclairMakeMeAdmin depending on ADMX

===

Name: "MakeMeAdmin Set Syslog-server"

Name: MakeMeAdmin ADMX - Set Syslog-server
Description: <None>
OMA-URI:`  ./Device/Vendor/MSFT/Policy/Config/MakeMeAdmin~Policy~makemeadmin:MakeMeAdmin~scc:SinclairRoot~SyslogServers`
Data type: String
Value: 
```
<enabled/>
<data id="SyslogMultiText" value="127.0.0.1"/>
```

As per the legend of applying a property from an already imported ADMX-file;
 ./Device/Vendor/MSFT/Policy/Config/{AppName}~{SettingType}~{CategoryPathFromADMX}/{SettingFromADMX}
{AppName} = MakeMeAdmin
{SettingType} = Policy
{CategoryPathFromADMX} = First for "SyslogServer" I have a "parentCategory ref" of:
makemeadmin:MakeMeAdmin
then moving "up" I have a "parentCategory ref" of:
scc:SinclairRoot
so all together
`makemeadmin:MakeMeAdmin~scc:SinclairRoot`
{SettingFromADMX} = SyslogServers

My reference has been this:
https://www.petervanderwoude.nl/post/deep-dive-ingesting-third-party-admx-files/

I have probably misunderstood something and would be helpful for someone explaining what I've done wrong, I guess I've "nestled" up in the hierarchy of the parentcategories the wrong way.

**Comment by leggo2** [comment]
@IwistIT Did you manage to get Azure AD joined devices to work with an Azure AD Security Group in "Allowed Entities" ?

**Comment by salihzett** [comment]
Has anyone how resolved this? maybe with an example for allowed entries and Intune?
I tried the script from [Oliver](https://github.com/okieselbach/Intune/blob/master/Convert-AzureAdObjectIdToSid.ps1) to convert AzureAD group ID to SID and set it to the ADMX entry, but it didn't work.

the user and the devices are azured joined (cloud environment)

**Comment by frost-cqre** [comment]
Has anyone figured out how to deploy this on AAD joined devices? I have a specific use case that anyone with valid company ID should be allowed to elevate admin rights, so I don't really care about the individual users. But I want to make sure that their admin time is limited and that local administrators group is regularly cleaned.

**Comment by pakbaetz** [comment]
hello @pseymour  Hi, I need your help with making myself an admin. I've followed the steps outlined, but I'm having trouble with the allowed entities settings. I've already deployed the OMA-URI settings in Intune, but every time I grant myself admin access, it doesn't seem to work. Although I receive a notification that I have admin access, I still can't perform admin tasks.

**Comment by pseymour** [comment]
Does your account end up in the Administrators group?

**Comment by pakbaetz** [comment]
yup, it was added in the admin group @pseymour , but after that, im still not having admin access. After I granted my device admin access in make me admin app, I tried to run the command prompt as admin, but it did not work. The admin access only works if rebooted the device.
